src/apint/utils.rs

```diff
@@ -62,16 +62,22 @@ pub(crate) enum ZipDataAccess<'a, 'b> {
 }
 
 #[derive(Debug, PartialEq, Eq)]
-pub(crate) enum ZipDataAccessMut<'a, 'b> {
+pub(crate) enum ZipDataAccessMutSelf<'a, 'b> {
     Inl(&'a mut Digit, Digit),
     Ext(&'a mut [Digit], &'b [Digit])
 }
 
+#[derive(Debug, PartialEq, Eq)]
+pub(crate) enum ZipDataAccessMutBoth<'a, 'b> {
+    Inl(&'a mut Digit, &'b mut Digit),
+    Ext(&'a mut [Digit], &'b mut [Digit])
+}
+
 // ============================================================================
 
 impl Width for ApInt {
     /// Returns the `BitWidth` of this `ApInt`.
-    #[inline]
+    #[inline(always)]
```

Comment:
Have you tested for the `#[inline(always)]` attribute? I have already seen reports not to use it too frequently and instead rely on the compiler for the simpler `#[inline]` in the general case.

Reply:
I overreached again now I realize. I searched it and found some interesting things I didn't know before. I am using it on internal functions that are used only a few times or on one-liners that probably only need `inline`. I think I will look at the perf a little more and remove some instances.

Reply:
As far as I have tested one should really only ever use `#[inline(always)]` for private one-liners that have been tested extensively about performance aspects for the certain use-case since `#[inline(always)]` is more or less pretty restrictive in its behaviour.